bundles/org.openhab.core.io.transport.mqtt/src/main/java/org/openhab/core/io/transport/mqtt/sslcontext/AcceptAllCertificatesSSLContext.java

```diff
@@ -31,6 +33,7 @@
  * @author David Graeff - Initial contribution
  */
 @Deprecated
+@NonNullByDefault
```

Comment:
Would it make more sense to remove deprecated classes now instead of adding annotations?

Reply:
The deprecated annotation was added in the process when we moved to the hivemq library.
Currently they are still in use in the public interface of the `MqttBrokerConnection`.
I think we agree, that we should refactor the `MqttBrokerConnection` into an interface and a factory. In this step we should definitly remove the deprecated functions /classes.

For this PR I was getting to use the SAT tools in depth for the first time and i wanted to see, if I was able to understand, what needs to be done for each warning. So, not needed here, but it was a side product from looking at all sat warnings for the project.

Reply:
Okay. I am fine with that.